src/interactable_spring.c

```diff
@@ -0,0 +1,309 @@
+
+static void initSprite_Interactable_Spring(u8, Interactable *, u16, u16, u8);
+static void Task_Interactable_Spring(void);
+static void sub_800E3D0(void);
+static bool32 sub_800E490(struct UNK_0808B3FC_UNK240 *p0, Interactable *inEntity,
+                          Sprite_Spring *spring, struct SomeStruct_59E0 *player);
+static void TaskDestructor_Interactable_Spring(struct Task *t);
+
+extern bool32 sub_800CDBC(struct UNK_0808B3FC_UNK240 *, s16, s16,
+                          struct SomeStruct_59E0 *);
+
+static const u16 sSpringAnimationData[NUM_SPRING_KINDS][SPRINGTYPE_COUNT][4] = {
+    {
+        { SA2_ANIM_SPRING, 0, 20, 0x0000 },
+        { SA2_ANIM_SPRING, 0, 20, 0x0800 },
+        { SA2_ANIM_SPRING, 2, 20, 0x0400 },
+        { SA2_ANIM_SPRING, 2, 20, 0x0000 },
+        { SA2_ANIM_SPRING, 4, 20, 0x0400 },
+        { SA2_ANIM_SPRING, 4, 20, 0x0000 },
+        { SA2_ANIM_SPRING, 4, 20, 0x0C00 },
+        { SA2_ANIM_SPRING, 4, 20, 0x0800 },
+        { SA2_ANIM_SPRING, 6, 15, 0x0400 },
+        { SA2_ANIM_SPRING, 6, 15, 0x0000 },
+    },
+    {
+        { SA2_ANIM_SPRING_MUS_PLA, 0, 20, 0x0000 },
+        { SA2_ANIM_SPRING_MUS_PLA, 0, 20, 0x0800 },
+        { SA2_ANIM_SPRING_MUS_PLA_2, 0, 20, 0x0400 },
+        { SA2_ANIM_SPRING_MUS_PLA_2, 0, 20, 0x0000 },
+        { SA2_ANIM_SPRING_MUS_PLA_2, 2, 25, 0x0400 },
+        { SA2_ANIM_SPRING_MUS_PLA_2, 2, 25, 0x0000 },
+        { SA2_ANIM_SPRING_MUS_PLA_2, 2, 25, 0x0C00 },
+        { SA2_ANIM_SPRING_MUS_PLA_2, 2, 25, 0x0800 },
+        { SA2_ANIM_SPRING_MUS_PLA_2, 4, 20, 0x0400 },
+        { SA2_ANIM_SPRING_MUS_PLA_2, 4, 20, 0x0000 },
+    },
+    {
+        { SA2_ANIM_SPIKES_TEC_BAS, 0, 20, 0x0000 },
+        { SA2_ANIM_SPIKES_TEC_BAS, 0, 20, 0x0800 },
+        { SA2_ANIM_SPIKES_TEC_BAS, 2, 20, 0x0400 },
+        { SA2_ANIM_SPIKES_TEC_BAS, 2, 20, 0x0000 },
+        { SA2_ANIM_SPIKES_TEC_BAS, 4, 16, 0x0400 },
+        { SA2_ANIM_SPIKES_TEC_BAS, 4, 16, 0x0000 },
+        { SA2_ANIM_SPIKES_TEC_BAS, 4, 16, 0x0C00 },
+        { SA2_ANIM_SPIKES_TEC_BAS, 4, 16, 0x0800 },
+        { SA2_ANIM_SPIKES_TEC_BAS, 6, 12, 0x0400 },
+        { SA2_ANIM_SPIKES_TEC_BAS, 6, 12, 0x0000 },
+    },
+};
+
+// Effects applied onto the player-state.
+// These trigger the player acceleration when touching each of the spring directions
+static const u8 gUnknown_080D53D0[SPRINGTYPE_COUNT]
+    = { 14, 15, 16, 17, 18, 19, 20, 21, 18, 19 };
+
+static const u16 sSpring_MusicPlant_Soundeffects[5]
+    = { SE_MUSIC_PLANT_SPRING_1, SE_MUSIC_PLANT_SPRING_2, SE_MUSIC_PLANT_SPRING_3,
+        SE_MUSIC_PLANT_SPRING_4, MUS_DUMMY };
+
+// Finished, but doesn't match.
+static void initSprite_Interactable_Spring(u8 springType, Interactable *inEntity,
+                                           u16 spriteRegionX, u16 spriteRegionY,
+                                           u8 spriteY)
+{
+    s16 springKind = SPRING_KIND_NORMAL;
+    struct Task *t = TaskCreate(&Task_Interactable_Spring, sizeof(Sprite_Spring), 0x2010,
```

Comment:
Btw you don't need to explicitly pass the functions as pointers as this is implicit

Reply:
It's just the style I got myself accustomed to. But if it's usually not done in your codebase, I'll let it be.

Reply:
Not done yet, I agree it's something which may provide some context, but for now I've not been passing with the `&`